android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -445,270 +449,301 @@ private MediaConstraints parseMediaConstraints(ReadableMap constraints) {
+                        return;
+                    }
 
-                if (audioSource != null) {
-                    String trackId = getNextTrackUUID();
-                    audioTrack
-                        = mFactory.createAudioTrack(trackId, audioSource);
                     if (audioTrack != null) {
-                        mMediaStreamTracks.put(trackId, audioTrack);
+                        mediaStream.addTrack(audioTrack);
+                        mMediaStreamTracks.put(audioTrackId, audioTrack);
 
                         WritableMap trackInfo = Arguments.createMap();
-                        trackInfo.putString("id", trackId);
+                        trackInfo.putString("id", audioTrackId);
                         trackInfo.putString("label", "Audio");
                         trackInfo.putString("kind", audioTrack.kind());
                         trackInfo.putBoolean("enabled", audioTrack.enabled());
-                        trackInfo.putString("readyState",
-                            audioTrack.state().toString());
+                        trackInfo.putString("readyState", audioTrack.state().toString());
+                        trackInfo.putBoolean("remote", false);
+                        tracks.pushMap(trackInfo);
+                    }
+
+                    if (videoTrack != null) {
+                        mediaStream.addTrack(videoTrack);
+                        mMediaStreamTracks.put(videoTrackId, videoTrack);
+
+                        WritableMap trackInfo = Arguments.createMap();
+                        trackInfo.putString("id", videoTrackId);
+                        trackInfo.putString("label", "Video");
+                        trackInfo.putString("kind", videoTrack.kind());
+                        trackInfo.putBoolean("enabled", videoTrack.enabled());
+                        trackInfo.putString("readyState", videoTrack.state().toString());
                         trackInfo.putBoolean("remote", false);
                         tracks.pushMap(trackInfo);
                     }
+
+                    Log.d(TAG, "mMediaStreamId: " + streamId);
+                    mMediaStreams.put(streamId, mediaStream);
+
+                    successCallback.invoke(streamId, tracks);
                 }
             }
-            if (audioTrack == null && audioConstraints != null) {
-                // FIXME The following does not follow the getUserMedia()
-                // algorithm specified by
-                // https://www.w3.org/TR/mediacapture-streams/#dom-mediadevices-getusermedia
-                // with respect to distinguishing the various causes of failure.
-                errorCallback.invoke(/* type */ null, "Failed to obtain audio");
-                return;
+        };
+
+        Context context = getReactApplicationContext();
+        for (String permission: requestPermissions) {
+            PermissionUtils.requestPermission(context, permission, callback);
+        }
```

Comment:
It works because the callback is designed to do the processing once all permission requests have finished. This will fire 2 activities and once both are completed the code will look at the results.

The `errorCallback` would be executed after all permissions were evaluated.